update-remote-repo.sh

```diff
@@ -1,57 +1,29 @@
-#!/usr/bin/env sh
+#!/bin/bash
+
+name=vm-archlinux
+archlinux_directory=qubes-packages-mirror-repo/$name
+package_directory=$archlinux_directory/pkgs
+db=qubes.db
 
 echo "Signing all unsigned packages"
 
-for filename in qubes-packages-mirror-repo/archlinux/pkgs/*.pkg.tar.xz ; do
+for filename in $package_directory/*.pkg.tar.xz ; do
 	if ! [ -f "$filename.sig" ] ; then
 		echo "Signing $filename"
-		gpg2 --detach-sign "$filename"
+		qubes-gpg-client-wrapper --detach-sign "$filename" > "$filename.sig"
 	fi
 done
 
-sudo mount --bind qubes-packages-mirror-repo/archlinux chroot-archlinux/tmp/qubes-packages-mirror-repo
-
-## release 3.1 Repository
-echo "Generating 3.1 repository"
-
-# Include 3.0 and 3.1 packages
-# Repo Add need packages to be added in the right order as it only keeps the last entered package version
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*3.0.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.1.db.tar.gz "$pkg";done;'
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*3.1.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.1.db.tar.gz "$pkg";done;'
-# Include XEN 4.6.0
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes-vm-xen-4.6.0*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.1.db.tar.gz "$pkg";done;'
-
-## release 3.2 Repository
-echo "Generating 3.2 repository"
-
-# Include 3.0 3.1 3.2 packages
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*3.0.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.2.db.tar.gz "$pkg";done;'
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*3.1.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.2.db.tar.gz "$pkg";done;'
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*3.2.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.2.db.tar.gz "$pkg";done;'
+sudo mount --bind $archlinux_directory chroot-$name/tmp/qubes-packages-mirror-repo
 
-# Include XEN 4.6.*
-sudo chroot chroot-archlinux su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes-vm-xen-4.6.*.pkg.tar.xz` ; do repo-add pkgs/qubes-r3.2.db.tar.gz "$pkg";done;'
-
-## release 4.0 Repository
 echo "Generating 4.0 repository"
+sudo chroot chroot-$name su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes*4.0.*.pkg.tar.xz` ; do repo-add pkgs/'"$db"'.tar.gz "$pkg";done;'
+sudo chroot chroot-$name su user -c 'cd /tmp/qubes-packages-mirror-repo; for pkg in `ls -v pkgs/qubes-vm-xen-4.8.*.pkg.tar.xz` ; do repo-add pkgs/'"$db"'.tar.gz "$pkg";done;'
```

Comment:
This is very fragile. Not all packages are guaranteed to have `4.0.*` version (those compatible with multiple Qubes releases, like split-gpg, don't follow this scheme). Generally, a single qubes-builder instance is supposed to deal with only a single Qubes version. So, much simpler and more reliable way would be to simply add all the packages. And use separate qubes-builder instances for different qubes versions.

Reply:
Modified it to be ``qubes*.pkg.tar.xz``, that should resolve that issue